src/probnum/filtsmooth/optim/_state_space_optimizer.py

```diff
@@ -2,14 +2,14 @@
 
 import abc
 
-from probnum.filtsmooth.optim import _stoppingcriterion
+from probnum.filtsmooth.optim import _stopping_criterion
```

Comment:
same here, simply import `FiltSmoothStoppingCriterion` directly.

Reply:
For some reason even without adding `StoppingCriterion` this change leads to cyclic imports.

Reply:
Note that this issue is already present on `main` and is not introduced by this PR. Likely there are some cyclic dependencies within the `filtsmooth` package causing this.

Reply:
Things like this should be caught by `pylint` once we enable it globally